src/events/sql.js

```diff
@@ -96,11 +96,15 @@ module.exports.wrapSqlQuery = function wrapSqlQuery(queryString, params, callbac
                 if (err) {
                     eventInterface.setException(dbapiEvent, err);
                 } else {
-                    let { rowCount } = res;
+                    let { rowCount, rows } = res;
                     if (!rowCount && res instanceof Array) {
                         rowCount = res.length;
+                        rows = res;
+                    }
+                    eventInterface.addToMetadata(dbapiEvent, { rowCount });                 
+                    if (rowCount && rows instanceof Array && rows.length) {
+                        eventInterface.addToMetadata(dbapiEvent, { 'sql.rows': rows });
```

Comment:
It's not good that there is no trimming here:
- When the maximum value is reached other (small) tags could be removed
- The `rows` object in many cases could be **very** big, and the process of handling it (before the trimming) could be wasteful

Reply:
Trying to trim it is very expensive if not needed. I tried this with 100 rows and it's fine, it's all data dependant. Either way the code is trimming before sending the trace

Reply:
I think that the code could be expensive before the trimming takes place

Reply:
Why? we just point the reference to an existing object. It's the same if we'll stringify 1M lines and trim it one-by-one (which will result in more stringify operations of less data) and then stringify the trace without 1M lines than stringify 1M+trace at the end.